tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/cocosnet_evaluator.py

```diff
@@ -0,0 +1,483 @@
+limitations under the License.
+"""
+
+from collections import OrderedDict
+from pathlib import Path
+import numpy as np
+import cv2
+
+from ..base_evaluator import BaseEvaluator
+from ..quantization_model_evaluator import create_dataset_attributes
+from ...adapters import create_adapter
+from ...config import ConfigError
+from ...data_readers import DataRepresentation
+from ...launcher import create_launcher
+from ...logging import print_info
+from ...preprocessor import PreprocessingExecutor
+from ...progress_reporters import ProgressReporter
+from ...representation import RawTensorPrediction, RawTensorAnnotation
+from ...utils import extract_image_representations, contains_all, get_path
+
+
+class CocosnetEvaluator(BaseEvaluator):
+    def __init__(
+            self, dataset_config, launcher, preprocessor_mask, preprocessor_image,
+            gan_model, check_model
+    ):
+        self.launcher = launcher
+        self.dataset_config = dataset_config
+        self.preprocessor_mask = preprocessor_mask
+        self.preprocessor_image = preprocessor_image
+        self.postprocessor = None
+        self.dataset = None
+        self.metric_executor = None
+        self.test_model = gan_model
+        self.check_model = check_model
+        self._metrics_results = []
+        self._part_by_name = {
+            'gan_network': self.test_model,
+            'verification_network': self.check_model
+        }
+
+    @classmethod
+    def from_configs(cls, config, delayed_model_loading=False):
+        launcher_config = config['launchers'][0]
+        dataset_config = config['datasets']
+
+        preprocessor_mask = PreprocessingExecutor(
+            dataset_config[0].get('preprocessing_mask')
+        )
+        preprocessor_image = PreprocessingExecutor(
+            dataset_config[0].get('preprocessing_image')
+        )
+        launcher = create_launcher(launcher_config, delayed_model_loading=True)
+
+        network_info = config.get('network_info', {})
+        if not delayed_model_loading:
+            cocosnet_network = network_info.get('cocosnet_network', {})
+            verification_network = network_info.get('verification_network', {})
+            model_args = config.get('_models', [])
+            models_is_blob = config.get('_model_is_blob')
+            if 'model' not in cocosnet_network and model_args:
+                cocosnet_network['model'] = model_args[0]
+                cocosnet_network['_model_is_blob'] = models_is_blob
+            if 'model' not in verification_network and model_args:
+                verification_network['model'] = model_args[1] if len(model_args) > 1 else None
+                verification_network['_model_is_blob'] = models_is_blob
+            network_info.update({
+                'cocosnet_network': cocosnet_network,
+                'verification_network': verification_network
+            })
+            if not contains_all(network_info, ['cocosnet_network', 'verification_network']):
```

Comment:
this condition contradicts with the next (you raise error if verification_network is not provided, while below you accept emptiness of this field)